app/controllers/saml_idp_controller.rb

```diff
@@ -148,10 +148,24 @@ def saml_response
     encode_response(
       current_user,
       authn_context_classref: requested_authn_context,
-      reference_id: auth_identity.session_uuid.gsub(/^_/, '')
+      reference_id: auth_identity.session_uuid.gsub(/^_/, ''),
+      encryption: saml_response_encryption_opts
     )
   end
 
+  def saml_response_encryption_opts
+    encryption_opts = nil
+    if current_user.last_identity
+      sp = ServiceProvider.new(current_user.last_identity.service_provider)
+      encryption_opts = {
+        cert: OpenSSL::X509::Certificate.new(sp.cert),
+        block_encryption: sp.block_encryption,
```

Comment:
Default makes sense.  Do we really want a choice at all here?  Maybe just mandate one until there is a clear for multiple?